test/select.spec.js

```diff
@@ -3378,4 +3378,52 @@ describe('ui-select tests', function () {
       expect(el.scope().$select.activeIndex).toBe(2);
     });
   });
+
+  describe('Header & Footer', function () {
+
+    var el;
+
+    function setupSelectComponent(theme, isMulti) {
```

Comment:
We have 2 functions for building up the template `createUitSelect` and `createUiSelectMultiple`.  I believe this functions can be changed to support theses tests.